pkg/local_object_storage/shard/control.go

```diff
@@ -127,7 +178,11 @@ func (s *Shard) Close() error {
 		components = append(components, s.writeCache)
 	}
 
-	components = append(components, s.pilorama, s.blobStor, s.metaBase)
+	components = append(components, s.pilorama, s.blobStor)
+
+	if s.GetMode() != ModeDegraded {
+		components = append(components, s.metaBase)
```

Comment:
We should still close it if the error was durieng `Init` stage, can we check if db is `nil` in `Close`?